transport/src/main/java/io/netty/channel/nio/NioEventLoop.java

```diff
@@ -302,11 +309,13 @@ public void run() {
     @Override
     protected void run() {
         for (;;) {
-            boolean oldWakenUp = wakenUp.getAndSet(false);
             try {
-                if (hasTasks()) {
-                    selectNow();
-                } else {
+                if (selectNow() == 0 && !hasTasks()) {
```

Comment:
yes that makes sense - I think it also allows us to get rid of the reset method since you have everything in scope you need. I'll refactor and update this PR.